doc/articles/Asynchronous Operations.md

```diff
@@ -0,0 +1,68 @@
+# Asynchronous Operations {#Asynchronous-Operations}
+
+Asynchronous operations in webgpu.h return a @ref WGPUFuture. This is an opaque handle which applications may use to poll or wait for completion of the asynchronous operation.
+
+On all implementations of webgpu.h, it must be possible to react to async operations without undue latency or spin-waiting.
+
+## Creation
+
+All asynchronous operations start when the application calls an asynchronous webgpu.h function. These functions return `WGPUFuture`, and take in their arguments a `CallbackInfo` struct like @ref WGPURequestDeviceCallbackInfo or @ref WGPUBufferMapCallbackInfo. `CallbackInfo` structs have the following members:
+ - a `WGPUChainedStruct const * nextInChain` pointer, for extensions.
+ - a @ref WGPUCallbackMode `mode` enum.
+ - a `callback` function pointer. For example:
+   ```typedef void (*WGPUBufferMapCallback)(
+        WGPUMapAsyncStatus status,
+        char const * message,
+        WGPU_NULLABLE void* userdata1,
+        WGPU_NULLABLE void* userdata2) WGPU_FUNCTION_ATTRIBUTE;```
+ - Two application-owned userdata members.\n
+   `void* userdata1`\n
+   `void* userdata2`
+
+The `callback` function pointer is called when the application _observes completion_ of the asynchronous operation. The `userdata1` and `userdata2` members are passed back to the application as the last two arguments in the callback function. Callbacks **may not** be called unless the application explicitly flushes them in order to _observe completion_. The point in time a callback is called depends on the @ref WGPUCallbackMode of the operation. webgpu.h provides three callback modes: `::WGPUCallbackMode_WaitAnyOnly`, `::WGPUCallbackMode_AllowProcessEvents`, and `::WGPUCallbackMode_AllowSpontaneous`.
```

Comment:
> explicitly flushes

Not true for AllowSpontaneous

Reply:
is AllowSpontaneous *guaranteed* to be spontaneous though? or is it just that it is allowed to happen? this is the reason why I wrote **may not**.

Reply:
oh, I think I misread "may not" as "must not" (it's ambiguous). Use "might not" or "may or may not"?

Reply:
used "might not"